admin_guide/registry_data_check_restore.adoc

```diff
@@ -0,0 +1,112 @@
+[[admin-guide-registry-data-check-restore]]
+= Image registry data check and restore
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+[IMPORTANT]
+====
+All that is described here is an experimental functionality and may
+change in the future.
+====
+
+== Overview
+
+Sometimes the data stored in the OpenShift Container Registry may differ
+from the data in the {product-title} cluster's etcd. It means that either
+the {product-title} cluster's database contains references on data that
+does not exist on the OpenShift Container Registry's storage, or the
+OpenShift Container Registry storage has data that is not referenced in
+the {product-title} cluster's database. This situation may occur as a result
+of an error or accidental deletion.
+
+There are two classes of problems:
+
+. The {product-title} cluster's database contains references to
+non-existent blobs. This usually happens due to storage damage. In this
+case, such imagestrems/images are not possible to use and any pull will
+end with error. The way to fix such images is to push them onto the
+OpenShift Container Registry again to restore the blobs. The problem is
+to understand which imagestreams were damaged.
+
+. Necessary images were removed from the {product-title} cluster's
+database, but remained in the OpenShift Container Registry storage. This
+may occur due to incorrect pruning settings or due to accidental manual
+removal. In this case, the problem is to restore the references in the
+{product-title} cluster's database in order to prevent blobs from being
+deleted during the hard pruning process.
+
+== Preparation
+
+Before start, you must give the registry's service account permissions to
+change the {product-title} cluster's database:
++
+----
+$ service_account=$(oc get -n default \
+    -o jsonpath=$'system:serviceaccount:{.metadata.namespace}:{.spec.template.spec.serviceAccountName}\n' \
+    dc/docker-registry)
+$ oc adm policy add-cluster-role-to-user cluster-admin ${service_account}
```

Comment:
there's no more minimal permission we can grant than cluster-admin?

Reply:
We need to list all imagestreams/images, get images and use imagestreammapping for any imagestream. I do not know what role it is guaranteed to provide this and is on all versions of the cluster. Maybe cluster-reader is enough, but I'm not sure.

Reply:
ok, this is fine for now.